File: 백준/Silver/2669. 직사각형 네개의 합집합의 면적 구하기/직사각형 네개의 합집합의 면적 구하기.java
import java.io.*;
import java.util.StringTokenizer;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[][] board = new int[101][101];
        for (int p = 0; p < 4; p++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int c1 = Integer.parseInt(st.nextToken());
            int r1 = Integer.parseInt(st.nextToken());
            int c2 = Integer.parseInt(st.nextToken());
            int r2 = Integer.parseInt(st.nextToken());
            for (int i = r1; i < r2; i++) {
                for (int j = c1; j < c2; j++) {
                    board[i][j]++;
                }
            }
        }
        int answer = 0;
        for (int i = 0; i < 101; i++) {
            for (int j = 0; j < 101; j++) {
                if (board[i][j] != 0) {
                    answer++;
                }
            }
        }
        System.out.println(answer);
    }
}

File: 백준/Silver/2669. 직사각형 네개의 합집합의 면적 구하기/직사각형 네개의 합집합의 면적 구하기.py
board = [[0] * 101 for _ in range(101)]

for _ in range(4):
    c1, r1, c2, r2 = map(int, input().split())
    for i in range(r1, r2):
        for j in range(c1, c2):
            board[i][j] += 1

answer = 0
for i in range(101):
    for j in range(101):
        if board[i][j] != 0:
            answer += 1

print(answer)